### ISSUE TITLE
Freeze when defining a class or function on the first line of a file

### ISSUE THREAD

**Issue by kalfa** [python-mode/python-mode#811]
It seems that python mode freezes vim when the content of a file is something like:

```
def foo():
    anything
```
or
```
class foo:
    anything
```

but not when:
```
anything
```
or
```
with foo:
    anything
```

I didn't try with anything else.
the code quote represent the entire file content, from the first line.
^C usually unblocks vim, which blocks immediately after if I keep on working on the same line (the first line of the class/function).

I can reproduce it with
```vi -u ~/.vim/bundle/python-mode/debug.vim a.py```

```
$ vim --version
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Apr 20 2017 14:21:26)
MacOS X (unix) version
Included patches: 1-567
Compiled by Homebrew
Huge version without GUI.  Features included (+) or not (-):
+acl             +file_in_path    +mouse_sgr       +tag_old_static
+arabic          +find_in_path    -mouse_sysmouse  -tag_any_white
+autocmd         +float           +mouse_urxvt     -tcl
-balloon_eval    +folding         +mouse_xterm     +termguicolors
-browse          -footer          +multi_byte      +terminfo
++builtin_terms  +fork()          +multi_lang      +termresponse
+byte_offset     -gettext         -mzscheme        +textobjects
+channel         -hangul_input    +netbeans_intg   +timers
+cindent         +iconv           +num64           +title
-clientserver    +insert_expand   +packages        -toolbar
+clipboard       +job             +path_extra      +user_commands
+cmdline_compl   +jumplist        +perl            +vertsplit
+cmdline_hist    +keymap          +persistent_undo +virtualedit
+cmdline_info    +lambda          +postscript      +visual
+comments        +langmap         +printer         +visualextra
+conceal         +libcall         +profile         +viminfo
+cryptv          +linebreak       +python          +vreplace
+cscope          +lispindent      -python3         +wildignore
+cursorbind      +listcmds        +quickfix        +wildmenu
+cursorshape     +localmap        +reltime         +windows
+dialog_con      -lua             +rightleft       +writebackup
+diff            +menu            +ruby            -X11
+digraphs        +mksession       +scrollbind      -xfontset
-dnd             +modify_fname    +signs           -xim
-ebcdic          +mouse           +smartindent     -xpm
+emacs_tags      -mouseshape      +startuptime     -xsmp
+eval            +mouse_dec       +statusline      -xterm_clipboard
+ex_extra        -mouse_gpm       -sun_workshop    -xterm_save
+extra_search    -mouse_jsbterm   +syntax
+farsi           +mouse_netterm   +tag_binary
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X_UNIX  -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L. -fstack-protector -L/usr/local/lib -L/usr/local/opt/libyaml/lib -L/usr/local/opt/openssl/lib -L/usr/local/opt/readline/lib  -L/usr/local/lib -o vim        -lncurses -liconv -framework Cocoa   -mmacosx-version-min=10.11 -fstack-protector-strong -L/usr/local/lib  -L/usr/local/Cellar/perl/5.24.1/lib/perl5/5.24.1/darwin-thread-multi-2level/CORE -lperl -lm -lutil -lc -F/usr/local/opt/python/Frameworks -framework Python   -lruby.2.4.1 -lobjc
```
and python-mode with the develop branch
```
~/.vim/bundle/python-mode$ git log -1
commit 01c3131343aaa3c76f8cb656b5e9f54ac90ca04b (HEAD -> develop, origin/develop, origin/HEAD)
Merge: 43c052e cc11bc3
Author: fmv1992 <fmv1992@gmail.com>
Date:   Mon Nov 13 23:34:33 2017 -0200

    Merge pull request #809 from alphaCTzo7G/patch-2

    Slack now has tonnes of unanswered question
```

let me know if more info is needed

**Comment by fmv1992** [comment]
This is a duplicate of:

https://github.com/python-mode/python-mode/issues/808

Closing here and will continue there.
